gltf/mesh.cpp

```diff
@@ -238,6 +238,9 @@ static bool canMergeMeshes(const Mesh& lhs, const Mesh& rhs, const Settings& set
 	if (lhs.streams.size() != rhs.streams.size())
 		return false;
 
+	if (lhs.extras.data && rhs.extras.data && strcmp(lhs.extras.data, rhs.extras.data))
```

Comment:
This comparison should only work if keep_extras is set. Also, this will not work if one of the mesh has empty extras. See material.cpp/areExtrasEqual for the correct way to implement this.